app/src/main/java/ca/pkay/rcloneexplorer/Services/SyncService.java

```diff
@@ -112,34 +117,45 @@ protected void onHandleIntent(@Nullable Intent intent) {
         startForeground(PERSISTENT_NOTIFICATION_ID_FOR_SYNC, builder.build());
 
         currentProcess = rclone.sync(remoteItem, remotePath, localPath, syncDirection);
+        JSONObject stats;
+        String notificationContent = "";
+        String[] notificationBigText = new String[5];
         if (currentProcess != null) {
             try {
                 BufferedReader reader = new BufferedReader(new InputStreamReader(currentProcess.getErrorStream()));
                 String line;
-                String notificationContent = "";
-                String[] notificationBigText = new String[5];
                 while ((line = reader.readLine()) != null) {
-                    if (line.startsWith("Transferred:") && !line.matches("Transferred:\\s+\\d+\\s+/\\s+\\d+,\\s+\\d+%$")) {
-                        String s = line.substring(12).trim();
-                        notificationBigText[0] = s;
-                        notificationContent = s;
-                    } else if (line.startsWith(" *")) {
-                        String s = line.substring(2).trim();
-                        notificationBigText[1] = s;
-                    } else if (line.startsWith("Errors:")) {
-                        notificationBigText[2] = line;
-                    } else if (line.startsWith("Checks:")) {
-                        notificationBigText[3] = line;
-                    } else if (line.matches("Transferred:\\s+\\d+\\s+/\\s+\\d+,\\s+\\d+%$")) {
-                        notificationBigText[4] = line;
-                    } else if (isLoggingEnable && line.startsWith("ERROR :")){
+                    JSONObject logline = new JSONObject(line);
+                    FLog.e("LOG", logline.toString());
+                    if(isLoggingEnable && logline.getString("level").equals("error")){
                         log2File.log(line);
+                    } else if(logline.getString("level").equals("warning")){
+                        //available stats:
+                        //bytes,checks,deletedDirs,deletes,elapsedTime,errors,eta,fatalError,renames,retryError
+                        //speed,totalBytes,totalChecks,totalTransfers,transferTime,transfers
+                        stats = logline.getJSONObject("stats");
+
+                        String speed = humanReadableBytes(stats.getLong("speed"))+"/s";
+                        String size = humanReadableBytes(stats.getLong("bytes"));
+                        String allsize = humanReadableBytes(stats.getLong("totalBytes"));
+                        double percent = ((double)  stats.getLong("bytes")/stats.getLong("totalBytes"))*100;
```

Comment:
With .optLong you should be able to specify a fallback in case the model property is not available. Alternatively, you could also use a proper jackson model.

Reply:
Yeah you are right - i need to make this more resilient. I will think of something to improve this including the exception

Reply:
Everything uses .optX now, except the "stats" object. 
This is a todo, but i am not yet sure what should happen if there is no stats object, i will add this later